python/HeaderService/hslib_salobj.py

```diff
@@ -671,11 +683,25 @@ async def announce(self, imageName):
 
         await self.evt_largeFileObjectAvailable.set_write(**kw)
         self.log.info(f"Sent {self.config.hs_name} largeFileObjectAvailable: {kw}")
+        self.completed_OK[imageName] = True
+
+    async def upload_to_s3(self, fileobj, key, imageName, nt=2):
+        s3upload = False
+        k = 1
+        while k <= nt and s3upload is False:
+            try:
+                await self.s3bucket.upload(fileobj=fileobj, key=key)
+                s3upload = True
+            except Exception as e:
+                self.log.error(f"Failed s3bucket.upload attempt # {k}/{nt} for: {imageName}")
+                self.log.exception(str(e))
+            k += 1
+        return s3upload
 
     def write(self, imageName):
         """ Function to call to write the header"""
         self.HDR[imageName].write_header(self.filename_HDR[imageName])
-        self.log.info("Wrote header to: {}".format(self.filename_HDR[imageName]))
+        self.log.info("Wrote header to filesystem: {}".format(self.filename_HDR[imageName]))
 
     def clean(self, imageName):
         """ Clean up imageName data structures"""
```

Comment:
Do you need to `del self.completed_OK[imageName]` here so that it doesn't become a memory leak?

Reply:
Yep. Thanks for noticing.